src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -101,75 +100,81 @@ def _find_timestamp_sequence(sequences, tokenizer, feature_extractor, max_source
         begin_idx = np.where(sequence == timestamp_begin)[0].item() if timestamp_begin in sequence else 0
         sequence = sequence[begin_idx:]
 
+        timestamp_tokens = sequence >= timestamp_begin
+        consecutive = np.where(timestamp_tokens[:-1] & timestamp_tokens[1:])[0] + 1
+        last_timestamp = np.where(timestamp_tokens)[0][-1]
+        consecutive = np.append(consecutive, last_timestamp) if last_timestamp not in consecutive else consecutive
         if seq_idx != 0:
             time -= stride_left + stride_right
             offset = int((time / feature_extractor.sampling_rate) / time_precision)
-            timestamp_tokens = np.where(sequence >= timestamp_begin)[0][1::2]
-            if len(timestamp_tokens) >= 1:
-                # if a big chunk lenght is used, we need to check all of the previous items
+            overlap_time = int((stride_left / feature_extractor.sampling_rate) / time_precision)
+            # relevant timestamps are the ones that are in the overlapping part
+            relevant_timestamp = np.where(sequence[consecutive] >= timestamp_begin + overlap_time)[0]
+            if relevant_timestamp.shape[0] > 0:
+                relevant_timestamp = (
+                    consecutive[relevant_timestamp[0] - 1] if relevant_timestamp[0] > 0 else consecutive[0]
+                )
+                # if a big stride is used, we need to check some of the previous items for the best overlap
                 best_match = 0
                 sliced_sequence = []
                 for idx, previous_sequence in enumerate(reversed(items)):
                     previous_tokens = previous_sequence[1:-1]
+                    if previous_sequence[0] < (timestamp_begin + offset - overlap_time) and idx != 0:
+                        break  # the previous sequence is too far in the past
```

Comment:
If we are looking too far in the past, we will go through the whole sequence. This cuts based on the possible overlapping timing data